### PR TITLE
v3: fix regression in aggregation

### PR DESCRIPTION
**Pull request by sougou**
I added a sanity check that disallowed distinct and grouping.
It turns out that such construct is in use in the wild. This
is an emergency fix to allow it back.

### CODE REVIEW

**Comment by sougou** [comment]
@michael-berlin I'll have to push this through soon since it caused a regression. May have to fix forward if you find issues.

File: data/test/vtgate/unsupported_cases.txt [hunk 1]
```diff
@@ -127,10 +127,6 @@
 "select distinct a, count(*) from user"
 "unsupported: distinct cannot be combined with aggregate functions"
 
-# distinct and group by
```
Comment (michael-berlin) [inline data/test/vtgate/unsupported_cases.txt]: Should you add a non-error test case for this? (If yes, I'm fine if you do this in a separate PR.)

Comment (sougou) [inline data/test/vtgate/unsupported_cases.txt]: Makes sense. I'll do a quick patch and push it through.
